promql/parser/parse.go

```diff
@@ -705,5 +705,34 @@ func (p *parser) addOffset(e Node, offset time.Duration) {
 	}
 
 	*endPosp = p.lastClosing
+}
+
+func (p *parser) setInstant(e Node, timestamp float64) {
```

Comment:
As elsewhere, avoid the word instant as it's confusing. For example this would mean there are instant range vectors.